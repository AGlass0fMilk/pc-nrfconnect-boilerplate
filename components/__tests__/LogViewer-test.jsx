import React from 'react';
import renderer from 'react-test-renderer';
import {
    decorateLogViewer,
    mapLogViewerDispatch,
    mapLogViewerState,
} from '../LogViewer';

describe('LogViewer', () => {
    it('should implement component decorator', () => {
        expect(decorateLogViewer).toBeInstanceOf(Function);
    });
    it('should implement dispatch mapper', () => {
        expect(mapLogViewerDispatch).toBeInstanceOf(Function);
    });
    it('should implement state mapper', () => {
        expect(mapLogViewerState).toBeInstanceOf(Function);
    });
});

describe('decorateLogViewer', () => {
    it('should render component', () => {
        const LogViewer = decorateLogViewer('div');
        expect(renderer.create(
            <LogViewer />,
        )).toMatchSnapshot();
    });
});
