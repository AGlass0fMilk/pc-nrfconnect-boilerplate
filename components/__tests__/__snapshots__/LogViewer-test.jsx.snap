exports[`decorateLogViewer should render component 1`] = `<div />`;
